app.py

```diff
@@ -43,12 +43,17 @@ def list_from(item):
 class ProductDBInterface:
     """Interface object so that the UI can easily obtain filtered products."""
     def parse_sql_tuple(self, x):
-        return (ScrapedProduct(name = x[2], source = x[5], price = x[3], product_type=x[9], photo = x[4], instock= x[6], new = x[7], price_check= x[8]))
+        return (ScrapedProduct(msrp = x[4], name = x[2], source = x[6], price = x[3], product_type=None, photo = x[5], instock= x[7], new = x[8], price_check= x[9]))
 
     @property
     def read_products_from_db(self):
         prod_tuple = db.getRetailers_Products()
         products = [self.parse_sql_tuple(x) for x in prod_tuple]
+        for i in products:
+            #fiddle with price string to convert to int
+            price = i.price[1:-3].replace(',', '')
+            i.savings = int(i.msrp) - int(price)
+            i.savings = "$" + str(i.savings)
         return products
```

Comment:
NB (non-blocking comment, i.e. you don't need to change this): 

I think we can probably separate this out into a helper function and clean it up a little bit:

```
def calc_and_set_savings_attr(self, product):
    price = product[1:-3].replace(',', '')
    product.savings = f"${int(product.msrp)} - int(price)}"
    return product

@property
def read_products_from_db(self):
    prod_tuple = db.getRetailers_Products()
    products = [self.parse_sql_tuple(x) for x in prod_tuple]
    products_with_savings = [self.calc_and_set_savings_attr(x) for x in products]
    return products_with_savings
```

Reply:
You might even be able to do something like this:

def calc_and_set_savings_attr(self, product):
    price = product[1:-3].replace(',', '')
    product.savings = f"${int(product.msrp)} - int(price)}"
    return product

@property
def read_products_from_db(self):
    return [self.calc_and_set_savings_attr(self.parse_sql_tuple(x)) for x in db.getRetailers_Products()]

Reply:
I would need to test that second option though.